### ISSUE TITLE
Printer forgets his commands when homing (starting a new print) or executing G28/G29 command 

### ISSUE THREAD

**Issue by lucianstefan** [MarlinFirmware/Marlin#6438]
Hey.

My printer started to restart after an incident  (i switched the power off when i was testing my Z calibration, and didn't want it to crush into the bed) 
Doing so, now when i try to print anything it restarts/resets when homing.
If i input into the command line the g-code G28/G29 it does the same thing, it restarts/resets when the Z axis goes down a few steps. ( I don't know if i am calling it right ? mybe it is a firmware crash)

The fact is that i can't use my printer anymore. And i don't know what to do.

I mounted a inductive probe a few days ago and had to recalibrate my Z axis and set the offset for it.

I am using 1.1.0 RC7 with ramps 1.4 and an original Arduino Mega.
My endstops are working fine, my probe is working fine when testing it with m119. Motors checked and work fine individually.

I don't know what to do :|

I am using Pronterface / Slc3r - Cura.

**Comment by Roxy-3D** [comment]
What type of bed leveling do you have active?

**Comment by lucianstefan** [comment]
Hello.

I am using Auto bed leveling - "AUTO_BED_LEVELING_FEATURE" - grid with 4 points. But it doesn't even get to probe a point because at the first one when going down it restarts/resets.

I can make a short video clip if that helps.

**Comment by thinkyhead** [comment]
First, upgrade to `RCBugFix` so that we are all on the same page. We might have fixed this already.

If the problem persists with `RCBugFix` then I would ask, what kind of microstepping do you have on your Z axis? What is your Z axis feed-rate? If you move the Z axis at top speed does the board reset?

**Comment by lucianstefan** [comment]
Thanks for trying to help!

Oki, i am running now on RC8. I assume this is what RCBugFix means? going to the latest version !?

The problem persists, but now in a different kind of way. My resets have gone away.

- these are my steps per unit: (X,Y,Z,E0)  81.2311, 79.00, 8036.00, 105.00
- feedrate: 300, 300, 4, 30

My new problem. 
Sending G28 - it's doing it well. 
Sending G29 - it's doing it well but not completely. After a fiew points that are probed at almost half of the way it freezez, and stays like that a while (30sec - 1 min) then in continues but it looks like it forgot his steps per unit... and when Z axis rises, he now rises like 50mm insted of 15mm and goes down slowley not fast and slow at the end, just slowley all the way.

Another example:
And if i send M114 and lets say we are at 10mm hight, and want to go to G0Z5 inputting this command doesn't do anything, it dosen't move. But if i then type M114 it says he is with Z axis at 5mm if then i type G0Z10 it goes at (what he thinks it's 10mm) so he actualy moves upper. So this is wired. Maybe Arduino board or RAMPS or a Z driver is malfunctioning?

**Comment by Bob-the-Kuhn** [comment]
RCBugFix is newer then RC8.  You can download it [here](https://github.com/MarlinFirmware/Marlin/tree/RCBugFix).

You'll need to manually move your machine specific items over to the new config files.

**Comment by lucianstefan** [comment]
Hello guys @Roxy-3D @Bob-the-Kuhn @thinkyhead  ....well it's me again.
I haven't powerd my printer from april. I just finished my Arhitecture exams.

And now i am back. Back with the same problem.

I've updated to version 1.1.x

I can issue g28 for homing, but when i try G29 for bed leveling, it just stays in the center and doesn't do enything. After a while it starts to probe 4 points, and at the fifth it stais again just... "thinking". After a minute it starts to probe point 5 but it doesn't know his Z hight anymore and probes in air. At point 6 it does the same thing.
It looks like he forgets his steps per unit, and goes nuts.
 I can't start any print :(

Software and hardware:
**ARDUINO ver. 1.8.3
MARLIN ver. 1.1.4**
Custom Prusa i3

Ramps 1.4
Arduino Mega 2560 R3 - original

**behavior 1:** homes normaly, it ABL's normaly - but when starting a print it forgets from the start there he is (Z10) and prints in air.
**behavior 2:** homes normaly,  it ABL's normaly half of the way then freezez for a minute or half, starts again to continue but it seams that he forgot his travel steps per unit. It travels 2 or 4 times more in hight, and at each point travels higher till he ends up stuck at the highest point.
**behavior 3:** homes normally, it ABL's normaly half of the way then skipes all the remaining points to measure and finishes. When print starts it prints in air. 

Each behavior was different after reseting - power ON/OFF the printer. Between them I haven't changed Marlin's settings. 

These are the main behaviors, it has some variations.
I have videos, but it seams the forum doesn't support them. If you know a site that does support temporary video upload tell me. 
**Here is my Config_h:**
[config_h.txt](https://github.com/MarlinFirmware/Marlin/files/1133940/config_h.txt)
